File: apps/web/lib/types/employees.ts
export interface Employee {
  id: string
  first_name: string
  last_name: string
  role: string
  email: string | null
  phone: string | null
  active: boolean
  supervisor_id: string | null
  created_at: string
}

export interface SupervisorRef {
  id: string
  first_name: string
  last_name: string
  role: string
}

export interface EmployeeDetail extends Employee {
  company_id: string
  supervisor: SupervisorRef | null
  updated_at: string
}

export interface Asset {
  id: string
  employee_id: string
  asset_type: 'car' | 'tool' | 'equipment' | 'other'
  name: string
  quantity: number
  unit: string | null
  serial_number: string | null
  notes: string | null
  active: boolean
  assigned_at: string
}

export interface CreateEmployeePayload {
  first_name: string
  last_name: string
  role: string
  email?: string | null
  phone?: string | null
  supervisor_id?: string | null
}

export interface UpdateEmployeePayload extends CreateEmployeePayload {}

export interface CreateAssetPayload {
  asset_type: string
  name: string
  quantity: number
  unit?: string | null
  serial_number?: string | null
  notes?: string | null
}

export interface UpdateAssetPayload extends CreateAssetPayload {}

export const ROLE_LABELS: Record<string, string> = {
  direktor: 'Direktor',
  inzenjer: 'Inženjer',
  administracija: 'Administracija',
  poslovoda: 'Poslovođa',
  radnik: 'Radnik',
}

export const ASSET_TYPE_LABELS: Record<string, string> = {
  car: 'Vozilo',
  tool: 'Alat',
  equipment: 'Oprema',
  other: 'Ostalo',
}

export const VALID_ROLES = ['direktor', 'inzenjer', 'administracija', 'poslovoda', 'radnik']
export const VALID_ASSET_TYPES = ['car', 'tool', 'equipment', 'other']

export function canManageEmployees(role: string): boolean {
  return ['direktor', 'inzenjer', 'administracija'].includes(role)
}

export function fullName(e: { first_name: string; last_name: string }): string {
  return `${e.first_name} ${e.last_name}`
}

File: services/api/dto/employees.go
package dto

import "time"

// ── Employee responses ────────────────────────────────────────────────────────

type EmployeeListItem struct {
	ID           string    `json:"id"`
	FirstName    string    `json:"first_name"`
	LastName     string    `json:"last_name"`
	Role         string    `json:"role"`
	Email        *string   `json:"email"`
	Phone        *string   `json:"phone"`
	Active       bool      `json:"active"`
	SupervisorID *string   `json:"supervisor_id"`
	CreatedAt    time.Time `json:"created_at"`
}

type SupervisorRef struct {
	ID        string `json:"id"`
	FirstName string `json:"first_name"`
	LastName  string `json:"last_name"`
	Role      string `json:"role"`
}

type EmployeeDetail struct {
	ID           string         `json:"id"`
	CompanyID    string         `json:"company_id"`
	FirstName    string         `json:"first_name"`
	LastName     string         `json:"last_name"`
	Role         string         `json:"role"`
	Email        *string        `json:"email"`
	Phone        *string        `json:"phone"`
	Active       bool           `json:"active"`
	SupervisorID *string        `json:"supervisor_id"`
	Supervisor   *SupervisorRef `json:"supervisor"`
	CreatedAt    time.Time      `json:"created_at"`
	UpdatedAt    time.Time      `json:"updated_at"`
}

// ── Employee requests ─────────────────────────────────────────────────────────

type CreateEmployeeRequest struct {
	FirstName    string  `json:"first_name" binding:"required"`
	LastName     string  `json:"last_name" binding:"required"`
	Role         string  `json:"role" binding:"required"`
	Email        *string `json:"email"`
	Phone        *string `json:"phone"`
	SupervisorID *string `json:"supervisor_id"`
}

type UpdateEmployeeRequest struct {
	FirstName    string  `json:"first_name" binding:"required"`
	LastName     string  `json:"last_name" binding:"required"`
	Role         string  `json:"role" binding:"required"`
	Email        *string `json:"email"`
	Phone        *string `json:"phone"`
	SupervisorID *string `json:"supervisor_id"`
}

// ── Asset responses ───────────────────────────────────────────────────────────

type AssetListItem struct {
	ID           string    `json:"id"`
	EmployeeID   string    `json:"employee_id"`
	AssetType    string    `json:"asset_type"`
	Name         string    `json:"name"`
	Quantity     float64   `json:"quantity"`
	Unit         *string   `json:"unit"`
	SerialNumber *string   `json:"serial_number"`
	Notes        *string   `json:"notes"`
	Active       bool      `json:"active"`
	AssignedAt   time.Time `json:"assigned_at"`
}

// ── Asset requests ────────────────────────────────────────────────────────────

type CreateAssetRequest struct {
	AssetType    string  `json:"asset_type" binding:"required"`
	Name         string  `json:"name" binding:"required"`
	Quantity     float64 `json:"quantity" binding:"required"`
	Unit         *string `json:"unit"`
	SerialNumber *string `json:"serial_number"`
	Notes        *string `json:"notes"`
}

type UpdateAssetRequest struct {
	AssetType    string  `json:"asset_type" binding:"required"`
	Name         string  `json:"name" binding:"required"`
	Quantity     float64 `json:"quantity" binding:"required"`
	Unit         *string `json:"unit"`
	SerialNumber *string `json:"serial_number"`
	Notes        *string `json:"notes"`
}
